tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -14,10 +15,40 @@
 from marionette_driver.errors import (NoSuchElementException,
                                       StaleElementException,
                                       InvalidResponseException)
+import mozfile
 
 from environment import GaiaTestEnvironment
 from file_manager import GaiaDeviceFileManager, GaiaLocalFileManager
 
+DEFAULT_SETTINGS = {
+    'airplaneMode.enabled': False,  # disable airplane mode
+    'audio.volume.alarm': 0,  # mute alarm audio
+    'audio.volume.content': 0,  # mute content audio
+    'audio.volume.notification': 0,  # mute audio notifications
+    'camera.sound.enabled': False,  # mute camera sounds
+    'edgesgesture.enabled': False,  # disable edge gestures
+    'ftu.manifestURL': None,  # disable the first time usage app
+    'keyboard.autocorrect': False,  # disable auto-correction of keyboard
+    'keyboard.clicksound': False,  # mute keyboard click sound
+    'keyboard.enabled-layouts': str({
+        'app://keyboard.gaiamobile.org/manifest.webapp': {
+            'en': True, 'number': True}}),  # reset keyboard layouts
+    'keyboard.vibration': False,  # disable keyboard vibration
+    'language.current': 'en-US',  # reset language to en-US
+    'lockscreen.enabled': False,  # disable lockscreen
```

Comment:
Rolling up conversation from IRC, turns out that this busts the passcode tests, since they didn't know the lockscreen was disabled and that hides all the passcode controls as well.

Per conclusions there, we need to add a fix to the passcode test to accommodate the disabled lockscreen. You've suggested they should just tap rather than using modifySettings() to reset the lock setting.

However,

https://github.com/mozilla-b2g/gaia/blob/master/tests/python/gaia-ui-tests/gaiatest/gaia_test.py#L693

...GaiaDevice.lock() is going to be broken by this assumption, since it locks by just turning the device off/on, and the entire set of lockscreen tests depend on that behavior. 

So they, at least should probably use modifySettings(), probably by defining a LockscreenTestCase() that calls it on setup and changing their own parent, so that we don't have to echo the behavior in every test file. I'd rather they didn't explicitly enable the lockscreen first, since that A) assumes the lockscreen works before we test the lockscreen works, and B) isn't what the user normally does.

If we're doing that there, I'd argue for just doing similar in the passcode test as well.

Either way, GaiaDevice.lock() needs to at least check whether the screen's disabled and assert out early if so. The exception would be if it can make an API call to lock whether or not we've disabled the screen, in which case I'd have it do that.

Of course, the other other option is we remove this part, and stick with unlocking the screen manually, and defer all this stuff to a followup bug where we take more time to design it than one line comment. :)